cardinal/ecs/component/component.go

```diff
@@ -14,11 +14,15 @@ type (
 		ID() TypeID
 		// New returns the marshaled bytes of the default value for the component struct.
 		New() ([]byte, error)
-		// Name returns the name of the component.
-		Name() string
 
-		Decode([]byte) (any, error)
 		Encode(any) ([]byte, error)
+		Decode([]byte) (any, error)
+		INameable
+	}
+
+	INameable interface {
```

Comment:
Instead of `Name()` I would just consolidate this with `ID()` that `IComponentType` currently has. 

I would also remove `SetID` since the user would already be setting it via their user-defined method. Also, not sure if possible, but I would prefer to consolidate these two interfaces because they seem to serve the same purpose.

Reply:
IComponent previously had a name. I just modularized it into a additional interface so that it can be public for a user. Name is not a new parameter, name is used extensively in the framework. 

I don't think it's good to expose ID to the user because it's a field that's sort of set internally. This has to be a unique  increasing number. And it's set dynamically in register components to ensure correctness.

Reply:
+1 with brian. In addition, this Name() is a part of CQL, and a human readable string is more desirable here.

Reply:
ok, aligned